helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -126,4 +123,53 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
     return hardConstraints.stream().map(HardConstraint::getDescription)
         .collect(Collectors.toList());
   }
+
+  // TODO investigate better ways to sort replicas. One option is sorting based on the creation time.
+  private List<AssignableReplica> getOrderedAssignableReplica(ClusterModel clusterModel) {
```

Comment:
As commented above, consider generalizing this to a pre-process step.

Reply:
For now, the greedy algorithm is the only one we need this ordering. So I don't think we should overcomplicate the interface before we have the more confirmed requirement.